### ISSUE TITLE
[DependencyInjection] GetBucketAsync fails due to Logger dependency

### ISSUE THREAD

**Issue by mtinnes** [brantburnett/Couchbase.Extensions#16]
Trying to call GetBucketAsync() results in the following. Logging should probably be left to the application and not the library: https://nblumhardt.com/2017/07/library-logging/

```

InvalidOperationException: Cannot create instance of type 'Microsoft.Extensions.Logging.ILoggerFactory' because it is either abstract or an interface.
Microsoft.Extensions.Configuration.ConfigurationBinder.CreateInstance(Type type)
Microsoft.Extensions.Configuration.ConfigurationBinder.BindInstance(Type type, object instance, IConfiguration config, BinderOptions options)
Microsoft.Extensions.Configuration.ConfigurationBinder.BindProperty(PropertyInfo property, object instance, IConfiguration config, BinderOptions options)
Microsoft.Extensions.Configuration.ConfigurationBinder.BindNonScalar(IConfiguration configuration, object instance, BinderOptions options)
Microsoft.Extensions.Configuration.ConfigurationBinder.BindInstance(Type type, object instance, IConfiguration config, BinderOptions options)
Microsoft.Extensions.Configuration.ConfigurationBinder.Bind(IConfiguration configuration, object instance, Action<BinderOptions> configureOptions)
Microsoft.Extensions.Configuration.ConfigurationBinder.Bind(IConfiguration configuration, object instance)
Microsoft.Extensions.Options.ConfigureNamedOptions<TOptions>.Configure(string name, TOptions options)
Microsoft.Extensions.Options.OptionsFactory<TOptions>.Create(string name)
Microsoft.Extensions.Options.OptionsManager<TOptions>+<>c__DisplayClass5_0.<Get>b__0()
System.Lazy<T>.ViaFactory(LazyThreadSafetyMode mode)
System.Lazy<T>.ExecutionAndPublication(LazyHelper executionAndPublication, bool useDefaultConstructor)
System.Lazy<T>.CreateValue()
System.Lazy<T>.get_Value()
Microsoft.Extensions.Options.OptionsCache<TOptions>.GetOrAdd(string name, Func<TOptions> createOptions)
Microsoft.Extensions.Options.OptionsManager<TOptions>.Get(string name)
Microsoft.Extensions.Options.OptionsManager<TOptions>.get_Value()
Couchbase.Extensions.DependencyInjection.Internal.ClusterProvider.GetClusterAsync()
System.Threading.Tasks.ValueTask<TResult>.get_Result()
System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable<TResult>+ConfiguredValueTaskAwaiter.GetResult()
Couchbase.Extensions.DependencyInjection.Internal.BucketProvider.<GetBucketAsync>b__4_0(string name)
```

**Comment by brantburnett** [comment]
Logging is being left to the application. In this case, the only dependency is on Microsoft.Extensions.Logging.Abstractions, which is the standard logging abstraction layer for .NET Core applications. This abstraction layer can be wired to most logging infrastructures, including Microsoft.Extensions.Logging, Serilog, and many others. The only requirement is that *some* logging framework which supports the abstraction layer be registered with DI.

However, this could probably stand to be better documented. This repo is actually in a bit of flux right now, as it is for the version compatible with SDK 2.x. For SDK 3.x, it has been moved into the main SDK repository as a first-class citizen: https://github.com/couchbase/couchbase-net-client/tree/master/src/Couchbase.Extensions.DependencyInjection.

I've filed an issue to track this documentation improvement: https://issues.couchbase.com/browse/NCBC-2560
